core/src/main/scala/cats/data/package.scala

```diff
@@ -1,33 +1,14 @@
 package cats
 
 package object data {
-  type NonEmptyList[A] = OneAnd[List, A]
   type NonEmptyStream[A] = OneAnd[Stream, A]
   type ValidatedNel[E, A] = Validated[NonEmptyList[E], A]
 
-  def NonEmptyList[A](head: A, tail: List[A] = Nil): NonEmptyList[A] =
-    OneAnd(head, tail)
-  def NonEmptyList[A](head: A, tail: A*): NonEmptyList[A] =
-    OneAnd[List, A](head, tail.toList)
-
   def NonEmptyStream[A](head: A, tail: Stream[A] = Stream.empty): NonEmptyStream[A] =
     OneAnd(head, tail)
   def NonEmptyStream[A](head: A, tail: A*): NonEmptyStream[A] =
     OneAnd(head, tail.toStream)
 
-  object NonEmptyList {
-    def fromReducible[F[_], A](fa: F[A])(implicit F: Reducible[F]): Eval[NonEmptyList[A]] =
```

Comment:
My main motivation for putting them in the type class as opposed to companion objects is to allow for efficient overrides. Having said that, I think it's actually probably pretty rare that you would want to ask for a `Reducible[F]` and then immediately call `.toNonEmptyList` on it. You can probably do whatever you want with the `Reducible` constraint (or another type class constraint that you add).

I don't feel too strongly about whether this should exist, so I don't mind taking it out if people would prefer. Though due to unfortunate binary compatibility with traits in Scala, it isn't one of those cases where "it's always easy to add it later" works :(